packages/backend-function/src/runtime/get_amplify_clients_configuration.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  GetObjectCommand,
+  NoSuchKey,
+  S3Client,
+  S3ServiceException,
+} from '@aws-sdk/client-s3';
+
+export type DataClientEnv = {
+  /* eslint-disable @typescript-eslint/naming-convention */
+  AMPLIFY_DATA_GRAPHQL_ENDPOINT: string;
+  AMPLIFY_DATA_MODEL_INTROSPECTION_SCHEMA_BUCKET_NAME: string;
+  AMPLIFY_DATA_MODEL_INTROSPECTION_SCHEMA_KEY: string;
+  AWS_ACCESS_KEY_ID: string;
+  AWS_SECRET_ACCESS_KEY: string;
+  AWS_SESSION_TOKEN: string;
+  AWS_REGION: string;
+  /* eslint-enable @typescript-eslint/naming-convention */
+};
+
+const isDataClientEnv = (env: unknown): env is DataClientEnv => {
+  return (
+    env !== null &&
+    typeof env === 'object' &&
+    'AMPLIFY_DATA_MODEL_INTROSPECTION_SCHEMA_BUCKET_NAME' in env &&
+    'AMPLIFY_DATA_MODEL_INTROSPECTION_SCHEMA_KEY' in env &&
+    'AWS_ACCESS_KEY_ID' in env &&
+    'AWS_SECRET_ACCESS_KEY' in env &&
+    'AWS_SESSION_TOKEN' in env &&
+    'AWS_REGION' in env &&
+    'AMPLIFY_DATA_GRAPHQL_ENDPOINT' in env &&
+    typeof env.AMPLIFY_DATA_MODEL_INTROSPECTION_SCHEMA_BUCKET_NAME ===
+      'string' &&
+    typeof env.AMPLIFY_DATA_MODEL_INTROSPECTION_SCHEMA_KEY === 'string' &&
+    typeof env.AWS_ACCESS_KEY_ID === 'string' &&
+    typeof env.AWS_SECRET_ACCESS_KEY === 'string' &&
+    typeof env.AWS_SESSION_TOKEN === 'string' &&
+    typeof env.AWS_REGION === 'string' &&
+    typeof env.AMPLIFY_DATA_GRAPHQL_ENDPOINT === 'string'
+  );
+};
+
+/* eslint-disable @typescript-eslint/naming-convention */
+export type ResourceConfig = {
+  API: {
+    GraphQL: {
+      endpoint: string;
+      region: string;
+      defaultAuthMode: string;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      modelIntrospection: any;
```

Comment:
It would not. The [ResourceConfig](https://github.com/aws-amplify/amplify-js/blob/main/packages/core/src/singleton/types.ts#L48-L57) type is full of optional inputs, but the [API type for graphql](https://github.com/aws-amplify/amplify-js/blob/main/packages/core/src/singleton/API/types.ts#L21-L48) starts to get pickier and the [MIS typing](https://github.com/aws-amplify/amplify-js/blob/main/packages/core/src/singleton/API/types.ts#L143-L153) is narrower still. I started reproducing enough typing to make this happy and ended up >100 loc of type before I decided it was better to advocate for a more permissive type in this case.

The experience with `Record<string, unknown>` is:
![image](https://github.com/user-attachments/assets/f5a03493-3594-4e2f-befa-69e2063fe8e8)

The experience with ` unknown` is:
![image](https://github.com/user-attachments/assets/c46b7e32-4919-4a18-8e66-73e9034fab65)

Reply:
My minimal type for MIS added 122 loc, and 22 types. Recommend keeping `any` and I have added a comment for context.

Reply:
To resolve this discussion. Offline we decided to keep this `any`.